api/sixtyfps-rs/lib.rs

```diff
@@ -281,40 +328,71 @@ pub trait ComponentHandle {
     fn run(&self);
 }
 
-/// Struct that's used to hold weak references for SixtyFPS components.
-pub struct Weak<T: ComponentHandle> {
-    inner: vtable::VWeak<re_exports::ComponentVTable, T::Inner>,
-}
-
-impl<T: ComponentHandle> Clone for Weak<T> {
-    fn clone(&self) -> Self {
-        Self { inner: self.inner.clone() }
+mod weak_handle {
+
+    use super::*;
+
+    /// Struct that's used to hold weak references of [SixtyFPS component](mod@crate#generated-components)
+    ///
+    /// In order to create a Weak, you should use [`ComponentHandle::as_weak`].
+    ///
+    /// Strongs references should not be captured by the functions given to a lambda,
+    /// as this would produce a reference loop and leak the component.
+    /// Instead, the callback function should capture a weak component.
+    ///
+    /// The Weak component also implement `Send` and can be send to another thread.
+    /// but the upgrade function will only return a valid component from the same thread
+    /// as the one it has been created from.
+    /// This is usefull to use with [`invoke_from_event_loop()`].
+    pub struct Weak<T: ComponentHandle> {
+        inner: vtable::VWeak<re_exports::ComponentVTable, T::Inner>,
+        thread: std::thread::ThreadId,
```

Comment:
I wonder if perhaps the order of these two should be swapped, so that on 32-bit systems the bigger field (64-bit thread id) comes first?